From 7169bea2b7c28fdbfbb3bf502bcad00ce61459a5 Mon Sep 17 00:00:00 2001
Message-ID: <7169bea2b7c28fdbfbb3bf502bcad00ce61459a5.1747779548.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1747779547.git.sharlatanus@gmail.com>
References: <cover.1747779547.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 20 May 2025 08:10:28 +0100
Subject: [PATCH 23/31] gnu: python-pint-pulsar: Update to 1.1.3.

* gnu/packages/astronomy.scm (python-pint-pulsar): Update to 1.1.3.
[arguments] <phases>: Add 'remove-deprecated-scripts.

Change-Id: I316e7d7e7ee2420f5067d6eb0d8be0de8e931f4f
---
 gnu/packages/astronomy.scm | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 6c13621880..96381acde9 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -6917,7 +6917,7 @@ (define python-asdf-wcs-schemas
 (define-public python-pint-pulsar
   (package
     (name "python-pint-pulsar")
-    (version "1.1.1")
+    (version "1.1.3")
     (source
      (origin
        (method git-fetch) ; no tests data in the PyPI tarball
@@ -6926,7 +6926,7 @@ (define-public python-pint-pulsar
              (commit version)))
        (file-name (git-file-name name version))
        (sha256
-        (base32 "0i8vznjgbdhrfv6kcq33zvsn6q40r29sscy8fmk13vx2zfwl8nf0"))))
+        (base32 "0n66z7v30q1981hx4q5vy3mibrh453gak8dd938038r5mwddya3f"))))
     (build-system pyproject-build-system)
     (arguments
      (list
@@ -6994,7 +6994,19 @@ (define-public python-pint-pulsar
               "tests/test_tcb2tdb.py"
               "tests/test_templates.py"
               "tests/test_variety_parfiles.py"
-              "tests/test_version.py")))
+              "tests/test_version.py")
+      #:phases
+      #~(modify-phases %standard-phases
+         (add-after 'unpack 'remove-deprecated-scripts
+           ;; ImportError: cannot import name 'CompositeMCMCFitter' from
+           ;; 'pint.mcmc_fitter'.
+           ;;
+           ;; Removed in 1.1.3: Broken fitter class `CompositeMCMCFitter`
+           ;; (this fitter was added seemingly to deal with combined radio and
+           ;; high-energy datasets, but has since been broken for a while.)
+           (lambda _
+             (substitute* "pyproject.toml"
+               (("event_optimize_multiple.*") "")))))))
     (native-inputs
      (list python-pytest
            python-setuptools
-- 
2.49.0

